File: find_pi.py
from math import pi
limit = True

def Precision(num, place):
    result = round(float(num), place)
    if place == 0:
        return int(result)
    else:
        return result

while limit:
    place = int(input("Enter how many decimal places of Pi to be displayed [LIMIT 15]: "))

    if place < 16 and place >= 0:
        limit = False
        result = Precision(pi, place)
    else:
        print("Decimal place out of bounds! Please enter a valid value.")

print("Pi in ", place, " decimal places is: ", result) 

File: find_pi.java
import java.lang.Math.*;
import java.math.BigDecimal;
import java.math.RoundingMode;
import java.util.Scanner;

class find_pi {
    public static void main(String[] args) {
        double pi = Math.PI;
        BigDecimal result;
        int place = 0;
        boolean limit = true;
        Scanner input = new Scanner(System.in);

        while (limit) {
            System.out.print("Enter how many decimal places of Pi to be displayed [LIMIT 15]: ");
            place = input.nextInt();

            if (place < 16 && place >= 0)
                limit = false;
            else
                System.out.println("Decimal place out of bounds! Please enter a valid value.");
        }

        result = BigDecimal.valueOf(pi).setScale(place, RoundingMode.HALF_UP);
        System.out.print("Pi in " + place + " decimal places is: " + result);
        input.close();
    }
}